client/ui/qml/Pages2/PageSettingsServerInfo.qml

```diff
@@ -35,156 +37,158 @@ PageType {
         anchors.fill: parent
 
-        spacing: 16
-
-        Repeater {
-            id: header
-            model: proxyServersModel
-
-            activeFocusOnTab: true
-            onFocusChanged: {
-                header.itemAt(0).focusItem.forceActiveFocus()
-            }
+        spacing: 4
 
-            delegate: ColumnLayout {
+        BackButtonType {
+            id: backButton
 
-                property alias focusItem: backButton
+            Layout.topMargin: 20
+            KeyNavigation.tab: headerContent.actionButton
 
-                id: content
+            backButtonFunction: function() {
+                if (nestedStackView.currentIndex === root.pageSettingsApiServerInfo &&
+                        root.processedServer.isCountrySelectionAvailable) {
+                    nestedStackView.currentIndex = root.pageSettingsApiLanguageList
+                } else {
+                    PageController.closePage()
+                }
+            }
+        }
 
-                Layout.topMargin: 20
+        HeaderType {
+            id: headerContent
+            Layout.fillWidth: true
+            Layout.leftMargin: 16
+            Layout.rightMargin: 16
 
-                BackButtonType {
-                    id: backButton
-                    KeyNavigation.tab: headerContent.actionButton
+            actionButtonImage: nestedStackView.currentIndex === root.pageSettingsApiLanguageList ? "qrc:/images/controls/settings.svg"
+                                                                                                 : "qrc:/images/controls/edit-3.svg"
 
-                    backButtonFunction: function() {
-                        if (nestedStackView.currentIndex === root.pageSettingsApiServerInfo &&
-                                ServersModel.getProcessedServerData("isCountrySelectionAvailable")) {
-                            nestedStackView.currentIndex = root.pageSettingsApiLanguageList
-                        } else {
-                            PageController.closePage()
-                        }
+            headerText: root.processedServer.name
+            descriptionText: {
+                if (root.processedServer.isServerFromGatewayApi) {
+                    if (nestedStackView.currentIndex === root.pageSettingsApiLanguageList) {
+                        return qsTr("Subscription is valid until ") + ApiServicesModel.getSelectedServiceData("endDate")
+                    } else {
+                        return ApiServicesModel.getSelectedServiceData("serviceDescription")
                     }
+                } else if (root.processedServer.isServerFromTelegramApi) {
+                    return root.processedServer.serverDescription
+                } else if (root.processedServer.hasWriteAccess) {
+                    return root.processedServer.credentialsLogin + " · " + root.processedServer.hostName
+                } else {
+                    return root.processedServer.hostName
                 }
```

Comment:
Should we use root.processedServer also in other places, for example TabBar and StackLayout below?

Reply:
https://github.com/amnezia-vpn/amnezia-client-internal/issues/61 in another task